### DISCUSSION TITLE
Modal for Vue-QRcode-Reader

### DISCUSSION THREAD

**Discussion by osahene** [quasarframework/quasar#12682] [General - Components / Directives / etc]
I am developing a mobile app with Quasar. When the Camera scans a QR code and then the RESULT is valid, I would want there to be a pop up (QDialog) and the message is written in the modal. 
I would be very grateful for your assistance.
Please find the code

`<template>
  <q-page class="flex flex-center">
    <div>
        Last result: <b>{{ result }}</b>
        
      <qrcode-stream @decode="onDecode"  @init="onInit"/>
      <q-btn label="Click me" color="primary" @click="showScanConfirmation = true" />
       <q-dialog v-show="showScanConfirmation" >
        <q-card class="my-card">
            <q-img src="https://cdn.quasar.dev/img/chicken-salad.jpg" />

            <q-card-section>
            <q-btn
                fab
                color="primary"
                icon="verified"
                class="absolute"
                style="top: 0; right: 12px; transform: translateY(-50%);"
            />

            <div class="row no-wrap items-center">
                <div class="col text-h6 ellipsis">
                
                </div>
                
            </div>

            <q-rating v-model="stars" :max="5" size="32px" />
            </q-card-section>

            <q-card-section class="q-pt-none">
            <div class="text-subtitle1">
                Name of product
            </div>
            <div class="text-caption text-grey">
                <p class="decode-result">
                    Last result: <b>{{ result }}</b>
                </p>
            </div>
            </q-card-section>

            <q-separator />

            <q-card-actions align="right">
            <q-btn v-close-popup flat color="primary" ripple label="Confirm" @click="secondDialog = true" />
            <q-btn v-close-popup flat class="bg-red" ripple label="Cancel" />
            </q-card-actions>
        </q-card>
        
       </q-dialog>
      

        

        <q-dialog v-model="secondDialog" persistent transition-show="scale" v-show="result ? true : false" transition-hide="scale">
            <q-card class="bg-teal text-white" style="width: 300px">
                <q-card-section>
                <div class="text-h6">Confirmation</div>
                </q-card-section>

                <q-card-section class="q-pt-none">

                    <div class="row no-wrap items-center">
                        <div class="col text-h6 ellipsis">
                        I agree to check out prodName 
                        </div>
                        <div class="col-auto text-green text-caption q-pt-md row no-wrap items-center">
                        <q-icon name="verified" />
                        </div>
                    </div>

                </q-card-section>

                <q-card-actions align="right" class="bg-white text-teal">
                <q-btn flat label="OK" v-close-popup class="q-mr-2" />
                <q-btn flat label="Cancel" v-close-popup="2" class="bg-blue" />
                </q-card-actions>
            </q-card>
        </q-dialog>

    </div>
  </q-page>

</template>

<script>
import { QrcodeStream } from "vue-qrcode-reader/src/index";
import { defineComponent} from 'vue';

export default defineComponent ({
components: {
    QrcodeStream,
    // QrcodeDropZone,
    // QrcodeCapture
  },
   data () {
    //  const options = [
    //   { text: "outline", value: this.paintOutline },
    // ]
    // const selected = options[1]
    return {
    //   persistent: ref(false),
      secondDialog: false,
      // camera: ref('auto'),
      result: null,
      showScanConfirmation: false,
      stars: 1,
      // selected
    }
  },
  methods: {

    paintOutline (detectedCodes, ctx) {
      for (const detectedCode of detectedCodes) {
        const [ firstPoint, ...otherPoints ] = detectedCode.cornerPoints

        ctx.strokeStyle = "red";

        ctx.beginPath();
        ctx.moveTo(firstPoint.x, firstPoint.y);
        for (const { x, y } of otherPoints) {
          ctx.lineTo(x, y);
        }
        ctx.lineTo(firstPoint.x, firstPoint.y);
        ctx.closePath();
        ctx.stroke();
      }
    },

    async onInit (promise) {
      try {
        await promise
      } catch (e) {
        console.error(e)
      } finally {
        this.showScanConfirmation = this.camera === "off"
      }
    },

    async onDecode (content) {
      this.result = content

      this.pause()
      await this.timeout(500)
      this.unpause()
    },

    unpause () {
      this.camera = 'auto'
    },

    pause () {
      this.camera = 'off'
    },

    timeout (ms) {
      return new Promise(resolve => {
        window.setTimeout(resolve, ms)
      })
    }
  }
})
</script>

<style>

</style>
`

**Comment by MilosPaunovic** [comment]
This is not a great way to post code - it's too incomprehensible.

Either create a Codepen (or any kind) of reproduction, or specify the exact problem that's troubling you. As I now understand, you want to listen to result from QRCode scanner and if it's valid, to show dialog? Eample for that you can find on the docs page, here: [https://quasar.dev/vue-components/dialog#basic](https://quasar.dev/vue-components/dialog#basic)

**Reply by osahene** [reply]
I am glad you know what I want to do. That is the exact thing. However, the dialog works with a button (click). I want it to pop up when the camera scans a QR code, to wit, not with a click. If you could assist me. Thank you.

**Comment by MilosPaunovic** [comment]
You can create a separate function that shows dialog, like in the Codepen below, and invoke it in the place you're listening to a successful scanning event.

[https://codepen.io/milospaunovic/pen/XWzwjqo?editors=1010](https://codepen.io/milospaunovic/pen/XWzwjqo?editors=1010)

**Reply by osahene** [reply]
I'm very grateful.
